src/docker-in-docker/install.sh

```diff
@@ -175,72 +177,67 @@ fi
-    docker_version_dot_escaped="${DOCKER_VERSION//./\\.}"
-    docker_version_dot_plus_escaped="${docker_version_dot_escaped//+/\\+}"
-    # Regex needs to handle debian package version number format: https://www.systutorials.com/docs/linux/man/5-deb-version/
-    docker_version_regex="^(.+:)?${docker_version_dot_plus_escaped}([\\.\\+ ~:-]|$)"
-    set +e # Don't exit if finding version fails - will handle gracefully
-        cli_version_suffix="=$(apt-cache madison ${cli_package_name} | awk -F"|" '{print $2}' | sed -e 's/^[ \t]*//' | grep -E -m 1 "${docker_version_regex}")"
-        engine_version_suffix="=$(apt-cache madison ${engine_package_name} | awk -F"|" '{print $2}' | sed -e 's/^[ \t]*//' | grep -E -m 1 "${docker_version_regex}")"
-    set -e
-    if [ -z "${engine_version_suffix}" ] || [ "${engine_version_suffix}" = "=" ] || [ -z "${cli_version_suffix}" ] || [ "${cli_version_suffix}" = "=" ] ; then
-        err "No full or partial Docker / Moby version match found for \"${DOCKER_VERSION}\" on OS ${ID} ${VERSION_CODENAME} (${architecture}). Available versions:"
-        apt-cache madison ${cli_package_name} | awk -F"|" '{print $2}' | grep -oP '^(.+:)?\K.+'
-        exit 1
-    fi
-    echo "engine_version_suffix ${engine_version_suffix}"
-    echo "cli_version_suffix ${cli_version_suffix}"
-fi
+# Helper to find appropriate package versions
+get_package_version_suffix() {
+    local package_name=$1
+    local package_version=$2
 
-# Version matching for moby-buildx
-if [ "${USE_MOBY}" = "true" ]; then
-    if [ "${MOBY_BUILDX_VERSION}" = "latest" ]; then
+    if [ "${package_version}" = "latest" ] || [ "${package_version}" = "lts" ] || [ "${package_version}" = "stable" ]; then
         # Empty, meaning grab whatever "latest" is in apt repo
-        buildx_version_suffix=""
+        echo ""
     else
-        buildx_version_dot_escaped="${MOBY_BUILDX_VERSION//./\\.}"
-        buildx_version_dot_plus_escaped="${buildx_version_dot_escaped//+/\\+}"
-        buildx_version_regex="^(.+:)?${buildx_version_dot_plus_escaped}([\\.\\+ ~:-]|$)"
-        set +e
-            buildx_version_suffix="=$(apt-cache madison moby-buildx | awk -F"|" '{print $2}' | sed -e 's/^[ \t]*//' | grep -E -m 1 "${buildx_version_regex}")"
+        # Fetch a valid version from the apt-cache (eg: the Microsoft repo appends +azure, breakfix, etc...)
+        # Regex needs to handle debian package version number format: https://www.systutorials.com/docs/linux/man/5-deb-version/
+        local package_version_dot_escaped="${package_version//./\\.}"
+        local package_version_dot_plus_escaped="${package_version_dot_escaped//+/\\+}"
+        local package_version_regex="^(.+:)?${package_version_dot_plus_escaped}([\\.\\+ ~:-]|$)"
+
+        set +e # Catch errors so we can show more useful output
+        local package_version_suffix="=$(apt-cache madison ${package_name} | awk -F"|" '{print $2}' | sed -e 's/^[ \t]*//' | grep -E -m 1 "${package_version_regex}")"
```

Comment:
Nit: Can we indent all the code in between `set` commands for better code readability?

Reply:
Sure thing. I typically don't consider `set +e` a scope, so I find the style a little confusing personally. Is there a bash styleguide this repo follows?

Reply:
> Is there a bash styleguide this repo follows?

I don't think there's a specific guide that we follow for this repo, trying to follow the historical stylings of these files (as well as happy to learn some new and standard styles) 😬